### PR TITLE
Fix: Shadow/Font size preset panel crashes the editor

### PR DESCRIPTION
**Pull request by t-hamano**
Temporary fix for #65711
Related to #64777

## What?
This PR fixes a critical error that occurs when returning to the previous screen from the shadow presets panel or the font presets panel.

This fix isn't ideal, but we need to prevent the issue from shipping in the upcoming Gutenberg 19.4.

## Why?

I found that the problem was caused by https://github.com/WordPress/gutenberg/pull/64777.

From [this comment](https://github.com/WordPress/gutenberg/issues/65711#issuecomment-2380625635):

> The cause of this problem is likely that in the component where the crash occurs, some logic relies on URL parameters, namely `useNavigator`.
> 
> For example, in the Shadows panel, `category` and `slug` are taken from URL parameters [here](https://github.com/WordPress/gutenberg/blob/faa6968936e8d996acafb8fd846d3ca08dad7088/packages/edit-site/src/components/global-styles/shadows-edit-panel.js#L75-L78).
> 
> However, when we return from this panel, the URL parameters have already been changed to new ones, so all parameters are `undefined`. The panel crashes because parameters are not expected to be `undefined`.

## How?

Perform an early return to avoid a critical error when the parameter is undefined. One ideal future solution is suggested in [this comment](https://github.com/WordPress/gutenberg/issues/65711#issuecomment-2383603543).

## Testing Instructions

- Open the site editor > global styles > Shadows > Choose one of the presets > click the Back button
- Open the site editor > global styles > Typography > Font size presets > Choose one of the presets > click the Back button

## Screenshots or screencast 

https://github.com/user-attachments/assets/013063ee-49f5-420c-8405-5627e26f3709

### CODE REVIEW

**Review by tyxla — APPROVED** [review]
This looks good 👍 Thanks for the fix!

Long-term, I agree with @ciampo that ideally the component would handle this - errors like these are consequences of keeping an open API with regards to those params.

**Review by ciampo — APPROVED** [review]
Agreed. LGTM as an urgent fix. I'll see if I can create a slightly more sophisticated fix , still in app-land — it will help us to understand how to handle this better at the `Navigator`-level.

The E2E failure doesn't seem related, and can be encountered across many other PRs.

**Comment by t-hamano** [comment]
Thanks for the review!

I think the E2E test failure is related to the core, so I would like to merge it once the core is fixed.

Core ticket: https://core.trac.wordpress.org/ticket/60666

**Comment by ciampo** [comment]
Thank you! I've also created a PR with an alternative solution that could avoid the "flash of empty screen" while transitioning out https://github.com/WordPress/gutenberg/pull/65790

**Comment by t-hamano** [comment]
Another critical error was found regarding font size presets: https://github.com/WordPress/gutenberg/issues/65774

I believe that the issue is not related to #64777, but simply a problem with the font size preset logic itself, and it also occurs in WP 6.7. On the other hand, this PR does not need to be backported to WP6.7.

Therefore, it might be better to merge this PR _after_ backporting the fix for #65774 to WP 6.7.

File: packages/edit-site/src/components/global-styles/font-sizes/font-size.js [hunk 1]
```diff
@@ -46,6 +46,10 @@ function FontSize() {
 
 	const [ globalFluid ] = useGlobalSetting( 'typography.fluid' );
 
+	if ( ! origin || ! slug ) {
```
Comment (ramonjd) [inline packages/edit-site/src/components/global-styles/font-sizes/font-size.js]: Oh nice! I promise I didn't steal this for https://github.com/WordPress/gutenberg/pull/65811

😆

**Review by ramonjd — APPROVED** [review]
Tested quickly in playground. Thanks for catching this so quickly.

**Comment by ciampo** [comment]
> Therefore, it might be better to merge this PR _after_ backporting the fix for #65774 to WP 6.7.

With https://github.com/WordPress/gutenberg/pull/65791 merged and cherry-picker, should we rebase, re-test and then merge this PR ?

**Comment by t-hamano** [comment]
I updated this PR as #65791 was merged.

The following warning was raised, so I moved the location of `useEffect` and merged the two early returns:

```
React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?
```

 The issue reported in #65711 has been resolved as before in this PR, and #65774 should not have reoccurred:

https://github.com/user-attachments/assets/ccbf5921-5abf-41cf-bef8-e3176dcb5371

**Review by ciampo — APPROVED** [review]
LGTM 🚀 

As mentioned before, the real fix should be applied in `Navigator`. Once that's done, we'll be able to rework the code, removing the early returns (and using `goTo` only as a fail-safe for other use-cases)

**Pull request merged**